src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -464,16 +468,24 @@ async Task SearchCoreAsync(IGrouping<INavigateToSearchService, Project> grouping
         // the dependency tree, which then pulls on N other projects, forcing results for this single project to pay
         // that full price (that would be paid when we hit these through a normal topological walk).
         //
-        // Note the projects in each 'dependency set' are already sorted in topological order.  So they will process
-        // in the desired order if we process serially.
-        var allProjects = _solution
+        // Note: the projects in each 'dependency set' are already sorted in topological order.  So they will process in
+        // the desired order if we process serially.
+        //
+        // Note: we should only process the projects that are in the ordered-list of projects the searcher is searching
+        // as a whole.
+        var filteredProjects = _solution
             .GetProjectDependencyGraph()
             .GetDependencySets(cancellationToken)
-            .SelectAsArray(s => s.SelectAsArray(_solution.GetRequiredProject));
+            .SelectAsArray(projectIdSet =>
+                projectIdSet.Where(id => allProjectIdSet.Contains(id))
+                            .Select(id => _solution.GetRequiredProject(id))
+                            .ToImmutableArray());
```

Comment:
the crash was that we we searching the entire solutio nfor SG docs, despite being asked for SG docs for the current project.  Thsi ultimately crashed becaues we sent notificatoins to the client of "we're done with ProjectX, we're done with ProjectY, ...", after telling the client "we're only searching one thign, so you should only get one 'done' notificatoin".  

Basically, our progress bar goes past 100% and crash happens :(